pkg/cache/fifo.go

```diff
@@ -89,6 +89,25 @@ func (c *FIFO) FromElems(key uint64) []*Item {
 	return elems
 }
 
+// FromLastSameElems returns continuous items that have the same comparable attribute with the the lastest one.
+func (c *FIFO) FromLastSameElems(checkFunc func(interface{}) bool, comparableFunc func(interface{}) string) []*Item {
```

Comment:
It is not friendly to use two function to check item. howa bout  checkFunc func(interface{})bool,string